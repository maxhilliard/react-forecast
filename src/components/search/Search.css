.searchContainer {
    composes: searchContainer from '../App.css';
    display: grid;
    grid-template-rows: 50px 50px;
    grid-template-columns: 100%;
    align-content: center;
    grid-template-areas:
        "searchInput"
        "searchButton";
}

.searchInput {
    border: 1px solid red;
    grid-area: searchInput;
}

.searchButton {
    border: 1px solid blue;
    grid-area: searchButton;
}