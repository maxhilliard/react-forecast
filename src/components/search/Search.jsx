import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SearchInput from './search-input/SearchInput';
import SearchButton from './search-button/SearchButton';

import { getForecast } from '../../services/forecast/forecastService';

import styles from './Search.css';

class Search extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchValue: '',
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleInputChange(e) {
        const { value: searchValue } = e.target;

        this.setState(prevState => ({
            ...prevState,
            searchValue,
        }));
    }

    async handleSubmit() {
        const { searchValue } = this.state;
        const { handleForecastResponse, handleErrorResponse } = this.props;

        try {
            const { list: forecast } = await getForecast(searchValue);
            handleForecastResponse(forecast);
        } catch (err) {
            handleErrorResponse();
        }
    }

    render() {
        const { searchValue } = this.state;

        return (
            <div className={styles.searchContainer}>
                <SearchInput handleInputChange={this.handleInputChange} value={searchValue} />
                <SearchButton isDisabled={!searchValue.length} handleSubmit={this.handleSubmit} />
            </div>
        );
    }
}

Search.propTypes = {
    handleForecastResponse: PropTypes.func.isRequired,
};

export default Search;
