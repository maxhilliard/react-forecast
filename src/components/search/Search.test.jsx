import React from 'react';
import { shallow } from 'enzyme';

import Search from './Search';

describe('Search', () => {
    describe('snapshot', () => {
        it('should match snapshot', () => {
            const componentShallow = shallow(<Search handleForecastResponse={() => {}} />);

            expect(componentShallow).toMatchSnapshot();
        });
    });
});
