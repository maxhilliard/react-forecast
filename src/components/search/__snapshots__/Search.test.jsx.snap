// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot should match snapshot 1`] = `
<div
  className="searchContainer"
>
  <SearchInput
    handleInputChange={[Function]}
    placeholder="Enter location"
    value=""
  />
  <SearchButton
    buttonText="Search"
    handleSubmit={[Function]}
    isDisabled={true}
  />
</div>
`;
